crates/core/src/operations/transaction/mod.rs

```diff
@@ -273,31 +280,28 @@ impl CommitData {
             commit_info.info = app_metadata.clone();
             actions.push(Action::CommitInfo(commit_info))
         }
-        Ok(CommitData {
+
+        for txn in &app_transactions {
+            actions.push(Action::Txn(txn.clone()))
+        }
+
+        CommitData {
             actions,
             operation,
             app_metadata,
-        })
+            app_transactions,
+        }
     }
 
-    /// Convert actions to their json representation
-    pub fn log_entry_from_actions<'a>(
-        actions: impl IntoIterator<Item = &'a Action>,
-    ) -> Result<String, TransactionError> {
+    /// Obtain the byte representation of the commit.
+    pub fn get_bytes(&self) -> Result<bytes::Bytes, TransactionError> {
         let mut jsons = Vec::<String>::new();
-        for action in actions {
+        for action in &self.actions {
             let json = serde_json::to_string(action)
                 .map_err(|e| TransactionError::SerializeLogJson { json_err: e })?;
             jsons.push(json);
         }
-        Ok(jsons.join("\n"))
-    }
-
-    /// Obtain the byte representation of the commit.
-    pub fn get_bytes(&self) -> Result<bytes::Bytes, TransactionError> {
-        // Data MUST be read from the passed `CommitData`. Don't add data that is not sourced from there.
-        let actions = &self.actions;
-        Ok(bytes::Bytes::from(Self::log_entry_from_actions(actions)?))
+        Ok(bytes::Bytes::from(jsons.join("\n")))
```

Comment:
since i to grok the commit logic a bit, it seemed like this was just seprated for legacy reasons, and the `log_entry_from_actions` was ever only supposed to be used with the data in `CommitData`